pkg/resource/grafana-datasource/grafana-loki-datasource.go

```diff
@@ -0,0 +1,111 @@
+package grafanadatasource
+
+import (
+	"encoding/base64"
+
+	"github.com/giantswarm/logging-operator/pkg/common"
+	loggedcluster "github.com/giantswarm/logging-operator/pkg/logged-cluster"
+	loggingcredentials "github.com/giantswarm/logging-operator/pkg/resource/logging-credentials"
+	"github.com/pkg/errors"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"sigs.k8s.io/yaml"
+)
+
+const (
+	lokiURL                   = "http://loki-gateway.loki.svc.cluster.local"
+	datasourceName            = "Loki"
+	datasourceSecretName      = "loki-datasource"
+	datasourceSecretNamespace = "monitoring"
+)
+
+type Values struct {
+	ApiVersion  int          `yaml:"apiversion" json:"apiversion"`
+	Datasources []datasource `yaml:"datasources" json:"datasources"`
+}
+
+type datasource struct {
+	Access         string         `yaml:"access" json:"access"`
+	Editable       bool           `yaml:"editable" json:"editable"`
+	BasicAuth      bool           `yaml:"basicAuth" json:"basicAuth"`
+	BasicAuthUser  string         `yaml:"basicAuthUser" json:"basicAuthUser"`
+	JsonData       jsonData       `yaml:"jsonData" json:"jsonData"`
+	Name           string         `yaml:"name" json:"name"`
+	Type           string         `yaml:"type" json:"type"`
+	Url            string         `yaml:"url" json:"url"`
+	SecureJsonData secureJsonData `yaml:"secureJsonData" json:"secureJsonData"`
+}
+
+type jsonData struct {
+	ManageAlerts bool `yaml:"manageAlerts" json:"manageAlerts"`
+}
+
+type secureJsonData struct {
+	BasicAuthPassword string `yaml:"basicAuthPassword" json:"basicAuthPassword"`
+}
+
+// ObservabilityBundleConfigMapMeta returns metadata for the observability bundle extra values configmap.
+func DatasourceConfigMapMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
```

Comment:
```suggestion
func DatasourceSecretMeta(lc loggedcluster.Interface) metav1.ObjectMeta {
```

Reply:
Done, including fixing name in the comment :sweat_smile: